storage/remote/storage.go

```diff
@@ -68,8 +68,21 @@ func (s *Storage) ApplyConfig(conf *config.Config) error {
 
 	// Update read clients
 	queryables := make([]storage.Queryable, 0, len(conf.RemoteReadConfigs))
-	for i, rrConf := range conf.RemoteReadConfigs {
-		c, err := NewClient(i, &ClientConfig{
+	for _, rrConf := range conf.RemoteReadConfigs {
+		hash, err := toHash(rrConf)
+		if err != nil {
+			return nil
+		}
+
+		// Set the queue name to the config hash if the user has not set
+		// a name in their remote write config so we can still differentiate
+		// between queues that have the same remote write endpoint.
+		name := string(hash[:6])
```

Comment:
I think the default value for the name should be the index. The index will convey much more information than a hash.

Reply:
I don't have a strong opinion. I don't tend to rely on the index.

Reply:
We actually shouldn't really use the index, since reordering the individual RW/RR configs in the config file will no longer result in restarts, but the index will have changed.

Reply:
If the index is used then reordering the config would need to result in restarting the queues. The restart-free reordering would only be available if a user specifies a name.

Reply:
One of the goals was to reduce the number of situations in which we restart queues when we don't really need to, so unless you or someone else feels really strongly about having the index as the default name then I think we should proceed as is.

Reply:
We chatted for awhile and decided to stick with the hash as the default for the name.